Sources/UDPClient.swift

```diff
@@ -111,6 +111,31 @@ open class UDPClient: Socket {
         }
     }
     
+    //TODO add multycast and boardcast
+    open func recv(_ expectlen: Int) -> ([Byte]?, String, Int) {
+        if let fd = self.fd {
+            var buff: [Byte] = [Byte](repeating: 0x0,count: expectlen)
+            var remoteipbuff: [Int8] = [Int8](repeating: 0x0,count: 16)
+            var remoteport: Int32 = 0
+            let readLen: Int32 = c_yudpsocket_recive(fd, buff: buff, len: Int32(expectlen), ip: &remoteipbuff, port: &remoteport)
+            let port: Int = Int(remoteport)
+            var address = ""
+            if let ip = String(cString: remoteipbuff, encoding: String.Encoding.utf8) {
+                address = ip
+            }
+          
+            if readLen <= 0 {
+                return (nil, address, port)
+            }
+          
+            let rs = buff[0...Int(readLen-1)]
+            let data: [Byte] = Array(rs)
+            return (data, address, port)
+        }
+        
+        return (nil, "no ip", 0)
```

Comment:
Maybe all data received should be represent in this struct?

    public struct ReceivedData<T> {
        let data: T?
        let result: Result
    }

Then this should be another PR..